client/cli/src/error.rs

```diff
@@ -18,61 +18,87 @@
 
 //! Initialization errors.
 
-
+use sp_core::crypto;
 
 /// Result type alias for the CLI.
 pub type Result<T> = std::result::Result<T, Error>;
 
 /// Error type for the CLI.
-#[derive(Debug, derive_more::Display, derive_more::From)]
+#[derive(Debug, thiserror::Error)]
 pub enum Error {
 	/// Io error
-	Io(std::io::Error),
+	#[error(transparent)]
+	Io(#[from] std::io::Error),
 	/// Cli error
-	Cli(structopt::clap::Error),
+	#[error(transparent)]
+	Cli(#[from] structopt::clap::Error),
 	/// Service error
-	Service(sc_service::Error),
+	#[error(transparent)]
+	Service(#[from] sc_service::Error),
 	/// Client error
-	Client(sp_blockchain::Error),
+	#[error(transparent)]
+	Client(#[from] sp_blockchain::Error),
 	/// scale codec error
-	Codec(parity_scale_codec::Error),
+	#[error(transparent)]
+	Codec(#[from] parity_scale_codec::Error),
 	/// Input error
-	#[from(ignore)]
+	#[error("Invalid input: {0}")]
 	Input(String),
 	/// Invalid listen multiaddress
-	#[display(fmt="Invalid listen multiaddress")]
-	#[from(ignore)]
+	#[error("Invalid listen multiaddress")]
 	InvalidListenMultiaddress,
-	/// Other uncategorized error.
-	#[from(ignore)]
+	/// Application specific error chain sequence forwarder.
+	#[error(transparent)]
+	Forwarded(#[from] Box<dyn std::error::Error>),
```

Comment:
```suggestion
	Application(#[from] Box<dyn std::error::Error>),
```

This or something else? `Forwarded` sounds weird.